pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -293,122 +349,97 @@ func (cmvt *cdcMixedVersionTester) timestampResolved() {
-	}
-}
-
-// assertValid checks if the validator has found any issues at the
-// time the function is called.
-func (cmvt *cdcMixedVersionTester) assertValid() versionStep {
-	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-		if failures := cmvt.validator.Failures(); len(failures) > 0 {
-			t.Fatalf("validator failures:\n%s", strings.Join(failures, "\n"))
-		}
-	}
-}
-
 // createChangeFeed issues a call to the given node to create a change
 // feed for the target table.
-func (cmvt *cdcMixedVersionTester) createChangeFeed(node int) versionStep {
-	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-		t.Status("creating changefeed")
-		db := u.conn(ctx, t, node)
+func (cmvt *cdcMixedVersionTester) createChangeFeed() func(context.Context, *logger.Logger, *rand.Rand, *mixedversion.Helper) error {
+	return func(ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+		// Wait for workload to be initialized so the target table exists.
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-cmvt.workloadInit:
+		}
+		_, db := h.RandomDB(r, cmvt.crdbNodes)
 
 		options := map[string]string{
 			"updated":  "",
 			"resolved": fmt.Sprintf("'%s'", resolvedInterval),
 		}
-		_, err := newChangefeedCreator(db, fmt.Sprintf("%s.%s", targetDB, targetTable), cmvt.kafka.sinkURL(ctx)).
+
+		_, err := newChangefeedCreator(db, fmt.Sprintf("%s.%s", targetDB, targetTable), cmvt.kafka.manager.sinkURL(ctx)).
 			With(options).
 			Create()
-		if err != nil {
-			t.Fatal(err)
+		if err == nil {
+			return err
 		}
-	}
-}
 
-func runCDCMixedVersions(
-	ctx context.Context, t test.Test, c cluster.Cluster, buildVersion *version.Version,
-) {
-	predecessorVersion, err := release.LatestPredecessor(buildVersion)
-	if err != nil {
-		t.Fatal(err)
+		return nil
 	}
+}
 
-	tester := newCDCMixedVersionTester(ctx, t, c)
-	tester.StartKafka(t, c)
-	defer tester.Cleanup()
-
-	rng, seed := randutil.NewPseudoRand()
-	t.L().Printf("random seed: %d", seed)
+// runWorkload runs the bank workload in the background.
+func (cmvt *cdcMixedVersionTester) runWorkload() func(context.Context, *logger.Logger, *rand.Rand, *mixedversion.Helper) error {
+	return func(ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+		bankRun := roachtestutil.NewCommand("%s workload run bank", test.DefaultCockroachPath).
+			Flag("max-rate", 10).
+			Arg("{pgurl%s}", cmvt.crdbNodes).
+			Option("tolerate-errors")
+
+		// Sanity check that the workload has been initlized.
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-cmvt.workloadInit:
+		}
 
-	// sqlNode returns the node to be used when sending SQL statements
-	// during this test. It is randomized, but the random seed is logged
-	// above.
-	sqlNode := func() int {
-		return tester.crdbNodes[rng.Intn(len(tester.crdbNodes))]
+		h.BackgroundCommand(bankRun.String(), option.NodeListOption{h.RandomNode(r, cmvt.crdbNodes)})
```

Comment:
`runWorkload()` is already called as a test `BackgroundFunc`; no need to create a `BackgroundCommand` here.